src/components/task/uc/deleteUserData.uc.unit.test.js

```diff
@@ -0,0 +1,332 @@
+const sinon = require('sinon');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const { ObjectId } = require('mongoose').Types;
+
+// const chaiAsPromised = require('chai-as-promised');
+const { deleteUserRelatedData } = require('./deleteUserData.uc');
+const repo = require('../repo/task.repo');
+const { toString: idToString } = require('../../../helper/compare').ObjectId;
+// const testObjects = require('../../../../test/services/helpers/testObjects');
+
+const { expect } = chai;
+const mapByName = (deleteUserDataFunctions) =>
+	deleteUserDataFunctions.reduce((map, fun) => {
+		map[fun.name] = fun;
+		return map;
+	}, {});
+
+chai.use(chaiAsPromised);
+
+const getExpectedUpdateMany = (modifiedDocuments) => ({ success: true, modifiedDocuments });
+const getExpectedDeleteMany = (deletedDocuments) => ({ success: true, deletedDocuments });
+
+describe.only('in "deleteUserData.uc" the function', () => {
+	describe('deleteUserData', () => {
+		const userId = new ObjectId();
+		const replaceUserId = new ObjectId();
+
+		// true result
+		it('should return a function that return an array.', async () => {
+			const result = deleteUserRelatedData();
+			expect(result).to.be.an('array').with.lengthOf(5);
+			expect(result[0], 'where first element is a function').to.be.a('function');
+			expect(result[1], 'where second element is a function').to.be.a('function');
+			expect(result[2], 'where third element is a function').to.be.a('function');
+			expect(result[3], 'where fourth element is a function').to.be.a('function');
+			expect(result[4], 'where fourth element is a function').to.be.a('function');
+		});
+
+		describe('deletePrivateSubmissions', () => {
+			const initStubs = ({ studentId }) => {
+				const deleteSingleSubmissionsFromUserStub = sinon.stub(repo, 'deleteSingleSubmissionsFromUser');
+				const privateSubmissions = [
+					{ _id: new ObjectId(), studentId },
+					{ _id: new ObjectId(), studentId },
+				];
+				deleteSingleSubmissionsFromUserStub.withArgs(userId).returns(getExpectedDeleteMany(2));
+
+				const findSingleSubmissionIdsByUserStub = sinon.stub(repo, 'findSingleSubmissionsByUser');
+				findSingleSubmissionIdsByUserStub.callsFake(() => []);
+				findSingleSubmissionIdsByUserStub.withArgs(userId).returns(privateSubmissions);
+
+				return { findSingleSubmissionIdsByUserStub, deleteSingleSubmissionsFromUserStub, privateSubmissions };
+			};
+
+			// true work
+			it('should return deleted private submissions', async () => {
+				// given
+				const {
+					findSingleSubmissionIdsByUserStub,
+					deleteSingleSubmissionsFromUserStub,
+					privateSubmissions,
+				} = initStubs({
+					studentId: userId,
```

Comment:
sure about sharing the userId between multiple tests? we just have stubs but without them  that might create conflicts.